### ISSUE TITLE
[YSQL] Support ALTER TABLE ADD PRIMARY KEY for colocated tables

### ISSUE THREAD

**Issue by m-iancu** [yugabyte/yugabyte-db#7534]
i.e. for tables in colocated databases or in a tablegroup.
Should be analogous to the regular table case.

See also: https://docs.yugabyte.com/latest/explore/colocated-tables.

One note about the regress tests needed:
* For colocated databases we cannot get the state from SQL so, for the regress tests, we can just verify that the alter works.
* For tablegroups though we can verify this by querying the respective system tables (see [yb_tablegroup.sql](https://github.com/yugabyte/yugabyte-db/blob/master/src/postgres/src/test/regress/sql/yb_tablegroup.sql#L15) tests). So we should verify in the tests that the new table is in the appropriate tablegroup using something like:
    ```
    SELECT s.relname, pg_tablegroup.grpname
        FROM (SELECT relname, unnest(reloptions) AS opts FROM pg_class) s, pg_tablegroup
        WHERE opts LIKE CONCAT('%tablegroup=', CAST(pg_tablegroup.oid AS text), '%');
    ```
cc @frozenspider

**Comment by frozenspider** [comment]
See also: #6159

**Comment by m-iancu** [comment]
Fixed by b318938.
